packages/flutter/lib/src/cupertino/text_theme.dart

```diff
@@ -47,97 +49,90 @@ const TextStyle _kDefaultTabLabelTextStyle = TextStyle(
   inherit: false,
   fontFamily: '.SF Pro Display',
   fontSize: 34.0,
   fontWeight: FontWeight.w700,
   letterSpacing: 0.41,
-  color: CupertinoColors.black,
+  color: CupertinoColors.label,
 );
 
-const TextStyle _kDefaultLargeTitleDarkTextStyle = TextStyle(
-  inherit: false,
-  fontFamily: '.SF Pro Display',
-  fontSize: 34.0,
-  fontWeight: FontWeight.w700,
-  letterSpacing: 0.41,
-  color: CupertinoColors.white,
-);
-
-// Eyeballed value since it's not documented in https://developer.apple.com/design/resources/.
-const TextStyle _kDefaultPickerLightTextStyle = TextStyle(
-  inherit: false,
-  fontFamily: '.SF Pro Display',
-  fontSize: 25.0,
-  fontWeight: FontWeight.w400,
-  letterSpacing: -0.41,
-  color: CupertinoColors.black,
-);
-
-// Eyeballed value since it's not documented in https://developer.apple.com/design/resources/.
-const TextStyle _kDefaultPickerDarkTextStyle = TextStyle(
+// Please update _DefaultCupertinoTextThemeData and _DefaultCupertinoTextThemeData
+// accordingly after changing the default color here, as their implementation
+// depends on the default value of the color field.
+//
+// Inspected on iOS 13 simulator with "Debug View Hierarchy".
+// Value extracted from off-center labels. Centered labels have a font size of 25pt.
+const TextStyle _kDefaultPickerTextStyle = TextStyle(
   inherit: false,
   fontFamily: '.SF Pro Display',
-  fontSize: 25.0,
+  fontSize: 21.0,
   fontWeight: FontWeight.w400,
   letterSpacing: -0.41,
-  color: CupertinoColors.white,
+  color: CupertinoColors.label,
 );
 
-// Eyeballed value since it's not documented in https://developer.apple.com/design/resources/.
+// Please update _DefaultCupertinoTextThemeData and _DefaultCupertinoTextThemeData
+// accordingly after changing the default color here, as their implementation
+// depends on the default value of the color field.
+//
 // Inspected on iOS 13 simulator with "Debug View Hierarchy".
-const TextStyle _kDefaultDateTimePickerLightTextStyle = TextStyle(
+// Value extracted from off-center labels. Centered labels have a font size of 25pt.
+const TextStyle _kDefaultDateTimePickerTextStyle = TextStyle(
   inherit: false,
   fontFamily: '.SF Pro Display',
   fontSize: 21,
   fontWeight: FontWeight.normal,
-  color: CupertinoColors.black,
+  color: CupertinoColors.label,
 );
 
-// Eyeballed value since it's not documented in https://developer.apple.com/design/resources/.
-// Inspected on iOS 13 simulator with "Debug View Hierarchy".
-const TextStyle _kDefaultDateTimePickerDarkTextStyle = TextStyle(
-  inherit: false,
-  fontFamily: '.SF Pro Display',
-  fontSize: 21,
-  fontWeight: FontWeight.normal,
-  color: CupertinoColors.white,
-);
+TextStyle _resolveTextStyle(TextStyle style, BuildContext context, bool nullOk) {
+  // This does not change the shadow color, etc.
+  Paint foreground, background;
+  if (style?.foreground != null)
+    foreground..color = CupertinoDynamicColor.resolve(style?.foreground?.color, context, nullOk: nullOk);
```

Comment:
I just realized this is not working because `Paint` is not storing the color as is:
```dart
  set color(Color value) {
    assert(value != null);
    final int encoded = value.value ^ _kColorDefault;
    _data.setInt32(_kColorOffset, encoded, _kFakeHostEndian);
  }
```
I'll get rid of `foreground` and `background`.

Reply:
Oh nice, sounds good!